src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -79,12 +79,31 @@ private static boolean isValidPath(Path filePath) {
     }
 
     /**
-     * Saves all data to this storage file.
-     *
-     * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * Saves all data to this storage file. If storage file was deleted, it will
+     * create the file and notify user.
+     * 
+     * @throws StorageOperationException
+     *             if there were errors converting and/or storing data to file.
+     * @throws FileNotFoundException
+     *             if file does not exist in set path.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+        final boolean fileExists = path.toFile().exists();
```

Comment:
- Why final? While running the program the file can go missing so fileExists value can change. For an immutable final variable it won't be possible to change the value.